### PR TITLE
Update feature controller to update featureset dependency

### PR DESCRIPTION
**Pull request by masudur-rahman**
Signed-off-by: Masudur Rahman <masud@appscode.com>

### CODE REVIEW

File: pkg/controllers/feature/feature_controller.go [hunk 1]
```diff
@@ -462,6 +466,105 @@ func (r *frReconciler) updateFeatureSetStatus(ctx context.Context, fs *uiapi.Fea
 	return client.IgnoreNotFound(err)
 }
 
+func (r *frReconciler) calculateFeatureSetDependency(ctx context.Context) error {
```
Comment (ArnobKumarSaha) [inline pkg/controllers/feature/feature_controller.go:469]: We are just not calculating. May be `updateAllFeatureSetDependancies()`

File: pkg/controllers/feature/feature_controller.go [hunk 2]
```diff
@@ -462,6 +466,105 @@ func (r *frReconciler) updateFeatureSetStatus(ctx context.Context, fs *uiapi.Fea
...
+	}
+	return nil
+}
+
+func (r *frReconciler) updateFeatureSetDependencyStatus(ctx context.Context, fsName string, enabled bool) error {
+	fs := &uiapi.FeatureSet{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: fsName}, fs)
+	if err != nil {
+		return err
+	}
```
Comment (ArnobKumarSaha) [inline pkg/controllers/feature/feature_controller.go:500]: We can move these dependancy related new changes to a seperate file.

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: pkg/controllers/feature/feature_controller.go (hunk 1 above)
Comment (masudur-rahman) [reply pkg/controllers/feature/feature_controller.go:469]: sounds good

File: pkg/controllers/feature/feature_controller.go (hunk 2 above)
Comment (masudur-rahman) [reply pkg/controllers/feature/feature_controller.go:500]: sounds good too

Comment (masudur-rahman) [reply pkg/controllers/feature/feature_controller.go]: moved

**Review by ArnobKumarSaha — APPROVED** [review]

**Pull request merged**